### PR TITLE
fix: don't show incorrect `extension` keyword completion

### PR DESCRIPTION
**Pull request by jkciesluk**
Maybe it would better to use something like `additionalFilter` instead of `reveresedTokensPredicate` for `extension`, but since there are only two keywords using that, it should be ok

fixes https://github.com/scalameta/metals/issues/4419

### CODE REVIEW

File: mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala [hunk 1]
```diff
@@ -81,7 +83,15 @@ object Keyword {
...
     Keyword("given", isBlock = true, isTemplate = true, isDefinition = true, isScala3 = true),
-    Keyword("extension", isBlock = true, isTemplate = true, isDefinition = true, isPackage = true, isScala3 = true),
+    Keyword(
+      "extension",
+      isBlock = true,
+      isTemplate = true,
+      isDefinition = true,
+      isPackage = true,
+      isScala3 = true,
+      reversedTokensPredicate = Some(!extendsPred(_))
```
Comment (jkciesluk) [inline mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala:93]: It's mostly about adding this, so we don't show `extension` where we would expect `extend`

File: mtags/src/main/scala-2/scala/meta/internal/pc/Keywords.scala [hunk 2]
```diff
@@ -20,14 +20,14 @@ trait Keywords { this: MetalsGlobal =>
 
     lazy val notInComment = checkIfNotInComment(pos, text, latestEnclosing)
 
-    lazy val reverseTokens: Iterator[Token] = {
+    lazy val reverseTokens: Array[Token] = {
```
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/Keywords.scala:23]: It's no longer reverse, no?
```suggestion
    lazy val tokens: Array[Token] = {
```

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/KeywordsCompletions.scala [hunk 3]
```diff
@@ -35,7 +35,7 @@ object KeywordsCompletions:
         val isParam = this.isParam(path)
         val isSelect = this.isSelect(path)
         lazy val text = completionPos.cursorPos.source.content.mkString
-        lazy val reverseTokens: Iterator[Token] =
+        lazy val reverseTokens: Array[Token] =
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/KeywordsCompletions.scala:38]:
```suggestion
        lazy val tokens: Array[Token] =
```

File: mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala [hunk 4]
```diff
@@ -66,8 +67,9 @@ case class Keyword(
       (this.isPackage && isPackage) ||
       (this.isMethodBody && isMethodBody) ||
       (this.isParam && isParam) ||
-      this.reversedTokensPredicate.exists(pred => pred(leadingReverseTokens))
-    }
+      (this.name == "extends" && predicate)
+    } &&
+    (if (this.name == "extension") predicate else true)
```
Comment (tgodzik) [inline mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala:72]:
```suggestion
    this.name != "extension" || predicate
```

File: mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala [hunk 5]
```diff
@@ -125,8 +135,11 @@ object Keyword {
     Keyword("then")
   )
 
-  private def extendsPred(leadingReverseTokens: Iterator[Token]): Boolean = {
-    leadingReverseTokens.filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF]).take(3).toList match {
+  private def extendsPred(leadingReverseTokens: Array[Token]): Boolean = {
+    leadingReverseTokens.reverseIterator
```
Comment (tgodzik) [inline mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala:139]: Why did you move reverse here actually?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the fix! I added a couple of comments.

File: mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala (hunk 5 above)
Comment (jkciesluk) [reply mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala]: It wasn't about moving reverse but changing Iterator to Array. Now we reverse in the same place as before, so here nothing changes

File: mtags/src/main/scala-2/scala/meta/internal/pc/Keywords.scala (hunk 2 above)
Comment (jkciesluk) [reply mtags/src/main/scala-2/scala/meta/internal/pc/Keywords.scala:23]: I've changed the place where we reverse everything back to where it was, so tokens are reversed again

**Review by tgodzik — APPROVED** [review]
LGMT!

**Review by tgodzik — CHANGES_REQUESTED** [review]
Och, actually some tests have started failing and adding extension additionally :O

**Review by tgodzik — APPROVED** [review]
LGTM!
